packages/frontend/editor-ui/src/stores/evaluation.store.ee.ts

```diff
@@ -0,0 +1,151 @@
+					}
+					acc[run.workflowId].push(run);
+					return acc;
+				},
+				{},
+			);
+		});
+
+		// Methods
+
+		const fetchTestCaseExecutions = async (params: { workflowId: string; runId: string }) => {
+			const testCaseExecutions = await evaluationsApi.getTestCaseExecutions(
+				rootStore.restApiContext,
+				params.workflowId,
+				params.runId,
+			);
+
+			testCaseExecutions.forEach((testCaseExecution) => {
+				testCaseExecutionsById.value[testCaseExecution.id] = testCaseExecution;
+			});
+
+			return testCaseExecutions;
+		};
+
+		// Test Runs Methods
+		const fetchTestRuns = async (workflowId: string) => {
+			loading.value = true;
+			try {
+				const runs = await evaluationsApi.getTestRuns(rootStore.restApiContext, workflowId);
+				runs.forEach((run) => {
+					testRunsById.value[run.id] = run;
+					if (['running', 'new'].includes(run.status)) {
+						startPollingTestRun(workflowId, run.id);
+					}
+				});
+				return runs;
+			} finally {
+				loading.value = false;
+			}
+		};
+
+		const getTestRun = async (params: { workflowId: string; runId: string }) => {
+			const run = await evaluationsApi.getTestRun(rootStore.restApiContext, params);
+			testRunsById.value[run.id] = run;
+			return run;
+		};
+
+		const startTestRun = async (workflowId: string) => {
+			const result = await evaluationsApi.startTestRun(rootStore.restApiContext, workflowId);
+			return result;
+		};
+
+		const cancelTestRun = async (workflowId: string, testRunId: string) => {
+			const result = await evaluationsApi.cancelTestRun(
+				rootStore.restApiContext,
+				workflowId,
+				testRunId,
+			);
+			return result;
+		};
+
+		const deleteTestRun = async (params: { workflowId: string; runId: string }) => {
+			const result = await evaluationsApi.deleteTestRun(rootStore.restApiContext, params);
+			if (result.success) {
+				const { [params.runId]: deleted, ...rest } = testRunsById.value;
+				testRunsById.value = rest;
+			}
+			return result;
+		};
+
+		// TODO: This is a temporary solution to poll for test run status.
+		// We should use a more efficient polling mechanism in the future.
+		const startPollingTestRun = (workflowId: string, runId: string) => {
+			const poll = async () => {
+				const run = await getTestRun({ workflowId, runId });
+				if (['running', 'new'].includes(run.status)) {
+					pollingTimeouts.value[runId] = setTimeout(poll, 1000);
+				} else {
+					delete pollingTimeouts.value[runId];
+				}
+			};
+			void poll();
+		};
+
+		const cleanupPolling = () => {
+			Object.values(pollingTimeouts.value).forEach((timeout) => {
+				clearTimeout(timeout);
+			});
+			pollingTimeouts.value = {};
+		};
+
+		return {
+			// State
+			fetchedAll,
+			testRunsById,
+			testCaseExecutionsById,
+
+			// Computed
+			isLoading,
+			isFeatureEnabled,
+			testRunsByWorkflowId,
+
+			// Methods
+			fetchTestCaseExecutions,
+			fetchTestRuns,
+			getTestRun,
+			startTestRun,
+			cancelTestRun,
+			deleteTestRun,
```

Comment:
these two functions are not used anymore. The endpoints exist. Should we keep all this dead code or clean it up as well?

Reply:
This is meant as an MVP to gather user feedback. I assume there will be a follow-up asap.